utils/audioprocessing/freesound_audio_analysis.py

```diff
@@ -18,143 +18,78 @@
 #     See AUTHORS file.
 #
 
-import errno
 import logging
 import os
 import shutil
-import signal
-import subprocess
-import sys
-import tempfile
 
 from django.conf import settings
 
 import utils.audioprocessing.processing as audioprocessing
+from utils.audioprocessing.freesound_audio_processing import FreesoundAudioProcessorBase
 from utils.audioprocessing.processing import AudioProcessingException
+from utils.filesystem import create_directories
 from utils.mirror_files import copy_analysis_to_mirror_locations
 
-
 logger = logging.getLogger("processing")
 
 
-def analyze(sound):
+class FreesoundAudioAnalyzer(FreesoundAudioProcessorBase):
 
-    def alarm_handler(signum, frame):
-        raise AudioProcessingException("timeout while waiting for Essentia")
+    def failure(self, message, error=None, failure_state="FA"):
```

Comment:
can the name of this be more like a verb? "set failure" or "set processing failed"?

Reply:
Addressed in 89e30d00ad951a2dfd2a69cb07fe65e3d6d2d748